extensions/utils/VaultClientWrapper.js

```diff
@@ -0,0 +1,145 @@
+const http = require('http');
+const https = require('https');
+const { ChainableTemporaryCredentials } = require('aws-sdk');
+const { errorUtils } = require('arsenal');
+
+const { authTypeAssumeRole } = require('../../lib/constants');
+const VaultClientCache = require('../../lib/clients/VaultClientCache');
+const CredentialsManager = require('../../lib/credentials/CredentialsManager');
+
+class VaultClientWrapper {
+    constructor(id, host, port, authConfig, logger) {
+        this._authConfig = authConfig;
+        this._transport = this._authConfig.transport;
+        this._clientId = id;
+        this._vaultHost = host;
+        this._vaultPort = port;
+        this.logger = logger;
+
+        if (this._transport === 'https') {
+            this.stsAgent = new https.Agent({ keepAlive: true });
+        } else {
+            this.stsAgent = new http.Agent({ keepAlive: true });
+        }
```

Comment:
This is quite nit picky and since we only have keep alive as an option may not be much better but we can do.
```suggestion
        const Agent = this._transport === 'https' ? https.Agent : http.Agent;
        this.stsAgent = new Agent({ keepAlive: true });
```